File: client/src/main/java/com/msc/serverbrowser/util/basic/MathUtility.java
package com.msc.serverbrowser.util.basic;

/**
 * @author Marcel
 * @since 27.01.2018
 */
public final class MathUtility {

	private MathUtility() {
		// Private constructor to prevent instantiation
	}

	/**
	 * Decreases or increases a number if its out of the given bounds.
	 *
	 * @param number the number to adjust
	 * @param lowerLimit lower limit that the number might get increased to
	 * @param upperLimit upper limit that the number might get decreased to
	 * @return the number or lowerLimit / upperLimit
	 */
	public static int limitUpperAndLower(final int number, final int lowerLimit, final int upperLimit) {
		if (lowerLimit > upperLimit) {
			throw new IllegalArgumentException("Lower bound can not be higher than upper bound (" + lowerLimit + "-" + upperLimit + ")");
		}

		return Math.max(Math.min(upperLimit, number), lowerLimit);
	}
}

File: client/src/main/kotlin/serverbrowser/util/basic/MathUtility.kt
package serverbrowser.util.basic

/**
 * @author Marcel
 * @since 27.01.2018
 */
object MathUtility {

    /**
     * Decreases or increases a number if its out of the given bounds.
     *
     * @param number the number to adjust
     * @param lowerLimit lower limit that the number might get increased to
     * @param upperLimit upper limit that the number might get decreased to
     * @return the number or lowerLimit / upperLimit
     */
    @JvmStatic
    fun limitUpperAndLower(number: Int, lowerLimit: Int, upperLimit: Int): Int {
        if (lowerLimit > upperLimit) {
            throw IllegalArgumentException("Lower bound can not be higher than upper bound ($lowerLimit-$upperLimit)")
        }

        return Math.max(Math.min(upperLimit, number), lowerLimit)
    }
}// Private constructor to prevent instantiation
